src/Products/GenericSetup/upgrade.py

```diff
@@ -123,7 +124,9 @@ class UpgradeEntity(object):
     """
     def __init__(self, title, profile, source, dest, desc, checker=None,
                  sortkey=0):
-        self.id = str(abs(hash('%s%s%s%s' % (title, source, dest, sortkey))))
+        id_base = f'{title}{source}{dest}{sortkey}'
```

Comment:
GenericSetup master still supports Python 2.7 and 3.5, so we cannot use f-strings.

Reply:
ops, ok, what a pity

Reply:
ok, fixed that